src/CompanyService.Data/CompanyUserRepository.cs

```diff
@@ -108,7 +99,7 @@ public Task<bool> DoesExistAsync(Guid userId)
 
     public async Task<bool> RemoveContractSubjectAsync(Guid contractSubjectId)
     {
-      List<DbCompanyUser> dbUsers = await _provider.CompaniesUsers.Where(x => x.ContractSubjectId == contractSubjectId).ToListAsync();
+      IQueryable<DbCompanyUser> dbUsers = _provider.CompaniesUsers.Where(x => x.ContractSubjectId == contractSubjectId);
```

Comment:
why u removed .ToListAsync()?

Reply:
before iterating in foreach data anyway downloaded from db, but is it async operation?

Reply:
> before iterating in foreach data anyway downloaded from db, but is it async operation?

I'll check this and revert if it is

Reply:
seems to be right